runtime/src/accounts_db.rs

```diff
@@ -1636,10 +1641,17 @@ impl AccountsDB {
                     )| {
                         if let Some((locked_entry, _)) = self.accounts_index.get(pubkey, None, None)
                         {
-                            locked_entry
+                            let is_alive = locked_entry
                                 .slot_list()
                                 .iter()
-                                .any(|(_slot, i)| i.store_id == *store_id && i.offset == *offset)
+                                .any(|(_slot, i)| i.store_id == *store_id && i.offset == *offset);
+                            if !is_alive {
+                                // If the account index entry was removed, that means no other
+                                // AppendVec's for this slot contains this pubkey, so it's safe
+                                // to unref
+                                locked_entry.unref()
```

Comment:
This comment confused me.

How about this:
This pubkey was found in the storage, but no longer exists in the index. It would have had a ref to the storage from the initial store, but it will not exist in the re-written slot. Unref it to keep the index consistent with removing the storage entries.

What if there are two dead entries in the slot for the same pubkey? Would they be unref'ed twice?